### ISSUE TITLE
Ensure + validate inventory `record_by_device_id` returns correct result

### ISSUE THREAD

**Issue by davidverweij** [ideafast/middleware-services#53]
In testing the BTF pipeline, I found the [inventory service](https://github.com/ideafast/middleware-services/blob/ff46d52953ca79c0d309e07adc015245d2782cf7/data_transfer/services/inventory.py#L29-L52) did not return the appropriate result. Potentially this has to do with the inventory not having an _end wear_ when the device is still being deployed. We should set up some tests to validate this method's functionality, and figure out why it doesn't pick up the query correctly.

### Test case
```python
record_by_device_id(
    "BTF-CS4P34",
    start=datetime.datetime(2021, 3, 26, 8, 30, 18, 370009), 
    end=datetime.datetime(2021, 3, 26, 8, 30, 21, 565704))
)
``` 
Returns: `None`

Logs do confirm that when `checkin` is null, [the method does replace it with the current time](https://github.com/ideafast/middleware-services/blob/ff46d52953ca79c0d309e07adc015245d2782cf7/data_transfer/services/inventory.py#L39-L41).

Manually querying the inventory returns:
```json
{
    "data": {
        "K-ABCABC": {
            "patient_id": "K-ABCABC",
            "device_id": "BTF-CS4P34",
            "checkout": "2021-03-18 19:38:49",
            "checkin": null
        },
       "etc" :{
        }
    },
    "meta": {
        "success": true,
        "errors": null
    }
}
```
Which indicates that is should have returned `K-ABCABC`. (anonymised)

**Comment by jawrainey** [comment]
For reference, this is likely becase [normalise_day](https://github.com/ideafast/middleware-services/blob/ff46d52953ca79c0d309e07adc015245d2782cf7/data_transfer/utils/__init__.py#L93-L97) only nulls hours and minutes, and not miliseconds. For example:

```python
>>> now
datetime.datetime(2021, 3, 26, 0, 0, 0, 647241)
>>> endr 
datetime.datetime(2021, 3, 26, 0, 0, 0, 565704)
>>> now <= endr
False
```

**Comment by jawrainey** [comment]
This was resolved in #56
